### PR TITLE
14449 Fixed Install Date format 

### PR DESCRIPTION
**Pull request by adam-govan**
made a new function to calculated seconds from Epoch and added logic to convert improperly stored times in milliseconds back to seconds.

### CODE REVIEW

File: SwrveSDK/SDK/Track/Swrve.m [hunk 1]
```diff
@@ -1815,7 +1816,9 @@ - (NSDictionary*) getDeviceProperties
 
     NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
     [dateFormatter setDateFormat:@"yyyyMMdd"];
-    NSDate *date = [NSDate dateWithTimeIntervalSince1970:self->install_time];
+    NSDate *date = [NSDate dateWithTimeIntervalSince1970:install_time];
+    NSString *dateString = [dateFormatter stringFromDate:date];
+    NSLog(@"Install Date String : %@", dateString);
```
Comment (Sergio-Mira) [inline SwrveSDK/SDK/Track/Swrve.m]: DebugLog?

File: SwrveSDK/SDK/Track/Swrve.m [hunk 2]
```diff
@@ -1815,7 +1816,9 @@ - (NSDictionary*) getDeviceProperties
...
     [deviceProperties setValue:[dateFormatter stringFromDate:date] forKey:@"swrve.install_date"];
```
Comment (Sergio-Mira) [inline SwrveSDK/SDK/Track/Swrve.m:1820]: dateString here instead of calling stringFormatDate again?

File: SwrveSDK/SDK/Track/Swrve.m (hunk 1 above)
Comment (adam-govan) [reply SwrveSDK/SDK/Track/Swrve.m]: Whoops! shouldn't be in there! 👍

File: SwrveSDK/SDK/Track/Swrve.m [hunk 3]
```diff
@@ -1950,13 +1966,21 @@ - (UInt64) getInstallTime:(NSString*)fileName withSecondaryFile:(NSString*)secon
...
     [currentTime writeToFile:fileName atomically:YES encoding:NSUTF8StringEncoding error:nil];
-    return (time / 1000 * 1000);
+    return time;
+}
+
+- (UInt64) secondsSinceEpoch {
+    return (unsigned long long)([[NSDate date] timeIntervalSince1970]);
+}
+
+- (UInt64) millisecondsSinceEpoch {
```
Comment (dominicmarmionswrve) [inline SwrveSDK/SDK/Track/Swrve.m]: Where is millisecondsSinceEpoch used?

File: SwrveSDK/SDK/Track/Swrve.m [hunk 4]
```diff
@@ -1950,13 +1966,21 @@ - (UInt64) getInstallTime:(NSString*)fileName withSecondaryFile:(NSString*)secon
...
-    NSString* currentTime = [NSString stringWithFormat:@"%llu", time/(UInt64)1000L];
+    UInt64 time = [self secondsSinceEpoch];
+    NSString* currentTime = [NSString stringWithFormat:@"%llu", time];
     [currentTime writeToFile:fileName atomically:YES encoding:NSUTF8StringEncoding error:nil];
-    return (time / 1000 * 1000);
+    return time;
+}
+
+- (UInt64) secondsSinceEpoch {
+    return (unsigned long long)([[NSDate date] timeIntervalSince1970]);
```
Comment (dominicmarmionswrve) [inline SwrveSDK/SDK/Track/Swrve.m:1979]: 👍

**Comment by dominicmarmionswrve** [comment]
LGTM
I think we need to test this by taking old sdk and upgrading to make sure it behaves as we expect.

File: SwrveSDK/SDK/Track/Swrve.m (hunk 3 above)
Comment (adam-govan) [reply SwrveSDK/SDK/Track/Swrve.m]: actually not currently. I was putting it in to provide an option but I'll take it out for now.

**Pull request merged**